carma-messenger-core/j2735_convertor/src/psm_convertor.cpp

```diff
@@ -0,0 +1,271 @@
+  out_msg.vert = ValueConvertor::valueJ2735ToCav<float>(
+      in_msg.vert, units::FIFTIETH_G_PER_M_PER_SEC_SQR, out_msg.presence_vector,
+      carma_v2x_msgs::msg::AccelerationSet4Way::ACCELERATION_VERTICAL_AVAILABLE,
+      j2735_v2x_msgs::msg::AccelerationSet4Way::ACCELERATION_VERTICAL_UNAVAILABLE);
+
+  // Convert Yaw Rate
+  out_msg.yaw_rate = ValueConvertor::valueJ2735ToCav<float>(
+      in_msg.yaw_rate, units::CENTI_DEG_PER_DEG, out_msg.presence_vector,
+      carma_v2x_msgs::msg::AccelerationSet4Way::YAWRATE_AVAILABLE, j2735_v2x_msgs::msg::AccelerationSet4Way::YAWRATE_UNAVAILABLE);
+}
+
+void PSMConvertor::convert(const j2735_v2x_msgs::msg::PositionalAccuracy& in_msg, carma_v2x_msgs::msg::PositionalAccuracy& out_msg)
+{
+  // Convert semi_major Axis
+  out_msg.semi_major = ValueConvertor::valueJ2735ToCav<float>(
+      in_msg.semi_major, units::TWENTIETH_M_PER_M, out_msg.presence_vector,
+      carma_v2x_msgs::msg::PositionalAccuracy::ACCURACY_AVAILABLE, j2735_v2x_msgs::msg::PositionalAccuracy::ACCURACY_UNAVAILABLE);
+
+  // Convert semi_minor Axis
+  out_msg.semi_minor = ValueConvertor::valueJ2735ToCav<float>(
+      in_msg.semi_minor, units::TWENTIETH_M_PER_M, out_msg.presence_vector,
+      carma_v2x_msgs::msg::PositionalAccuracy::ACCURACY_AVAILABLE, j2735_v2x_msgs::msg::PositionalAccuracy::ACCURACY_UNAVAILABLE);
+
+  // Convert Orientation
+  out_msg.orientation = ValueConvertor::valueJ2735ToCav<double>(
+      in_msg.orientation, units::DEG_360_OVER_65535_PER_DEG, out_msg.presence_vector,
+      carma_v2x_msgs::msg::PositionalAccuracy::ACCURACY_ORIENTATION_AVAILABLE,
+      j2735_v2x_msgs::msg::PositionalAccuracy::ACCURACY_ORIENTATION_UNAVAILABLE);
+};
+
+void PSMConvertor::convert(const j2735_v2x_msgs::msg::PathHistory& in_msg, carma_v2x_msgs::msg::PathHistory& out_msg)
+{
+    out_msg.curr_gnss_status = in_msg.curr_gnss_status;
+    out_msg.initial_position.lat.latitude = in_msg.initial_position.lat.latitude * latitude_conversion_const_;
+    out_msg.initial_position.lon.longitude = in_msg.initial_position.lon.longitude * longitude_conversion_const_;
+    out_msg.initial_position.elevation.elevation = in_msg.initial_position.elevation.elevation * elevation_conversion_const_;
+    out_msg.initial_position.utc_time = in_msg.initial_position.utc_time;
+    out_msg.initial_position.time_confidence = in_msg.initial_position.time_confidence;
+    out_msg.initial_position.pos_confidence = in_msg.initial_position.pos_confidence;
+    out_msg.initial_position.speed_confidence = in_msg.initial_position.speed_confidence;
+
+    convert(in_msg.initial_position.heading, out_msg.initial_position.heading);
+    convert(in_msg.initial_position.speed.speed, out_msg.initial_position.speed.speed);
+    convert(in_msg.initial_position.pos_accuracy, out_msg.initial_position.pos_accuracy);
+
+    for (auto i : out_msg.crumb_data.points) {
+
+        carma_v2x_msgs::msg::PathHistoryPoint j;
+
+        j.lat_offset.offset = i.lat_offset.offset * 1e-7;
```

Comment:
Looks like these constants have been defined in the header, you can use those instead

Reply:
where in the header ? i dont see them in the header file